tests/outbound-wasi-http-v0.2.0-no-perms/test.json5

```diff
@@ -0,0 +1,30 @@
+{
+    "invocations": [
+        {
+            "request": {
+                "path": "/",
+                "headers": [
+                    {
+                        "name": "url",
+                        "value": "http://localhost:%{port=80}"
+                    }
+                ]
+            },
+            "response": {
+                "status": 500,
+                "headers": [
+                    {
+                        "name": "transfer-encoding",
+                        "value": "chunked"
+                    },
+                    {
+                        "name": "Date",
+                        "optional": true
```

Comment:
Does this imply that this header check is exhaustive? I wonder if that's necessary or if we'd be better off just checking for certain headers (perhaps with e.g. `null` meaning "reject if this header exists").

:shrug:

Reply:
Yes, the header check is exhaustive. I'd like to stick with exhaustive checks for now, but I understand that as we try to document what runtimes are and are not allowed to do we might have to be less restrictive in the future.

Reply:
I feel quite confident that runtimes won't return all the same headers, but waiting is fine.

Reply:
Actually, I think the `transfer-encoding` example above is a good one here; it would have been semantically the same to return with no `transfer-encoding` (under certain circumstances...).

Reply:
https://github.com/fermyon/conformance-tests/issues/3